srcCxx/shape_main.cxx

```diff
@@ -732,12 +974,44 @@ class ShapeApplication {
         if (dp)  dp->delete_contained_entities( );
         if (dpf) dpf->delete_participant( dp );
 
+        free(topics);
```

Comment:
Is there a particular reason you are using malloc/free as opposed to new/delete with STL containers?

Reply:
no specific reasons for using the C API